pkg/tests/tasks/security/authorization/jwt_test.go

```diff
@@ -18,157 +18,196 @@ import (
-		if !strings.Contains(msg, "403") {
-			log.Log.Errorf("Verify request without groups claim JWT Unexpected response: %s", msg)
-			t.Errorf("Verify request without groups claim JWT Unexpected response: %s", msg)
-		} else {
-			log.Log.Infof("Success. Get expected response: %s", msg)
-		}
+		t.NewSubTest("Security authorization allow JWT requestPrincipal").Run(func(t test.TestHelper) {
+			t.Cleanup(func() {
+				oc.DeleteFromString(t, ns, JWTRequireRule)
+			})
+			oc.ApplyString(t, ns, JWTRequireRule)
+
+			t.LogStep("Verify that a request with a valid JWT is allowed")
+			assertRequestJWTAccepted(t, ns, curl_JWT("Authorization: Bearer %s", token))
+
+			t.LogStep("Verify request without a JWT is denied")
+			assertRequestJWTDenied(t, ns, curl_JWT())
+
+		})
+
+		t.NewSubTest("Security authorization allow JWT claims group").Run(func(t test.TestHelper) {
+			t.Cleanup(func() {
+				oc.DeleteFromString(t, ns, JWTGroupClaimRule)
+			})
+			oc.ApplyString(t, ns, JWTGroupClaimRule)
+
+			t.LogStep("Verify that a request with the JWT that includes group1 in the groups claim is allowed")
+			assertRequestJWTAccepted(t, ns, curl_JWT("Authorization: Bearer %s", tokenGroup))
+
+			t.LogStep("Verify that a request with a JWT, which does not have the groups claim is rejected")
+			assertRequestJWTDenied(t, ns, curl_JWT("Authorization: Bearer %s", token))
+		})
+	})
+}
+
+func curl_JWT(headers ...string) string {
+	headerArgs := ""
+	for _, header := range headers {
+		headerArgs += fmt.Sprintf(` -H "%s"`, header)
+	}
+	return fmt.Sprintf(`curl "http://httpbin:8000/headers" -sS -o /dev/null %s -w "%%%%{http_code}\n"`, headerArgs)
+}
+
+func assertRequestJWTDenied(t test.TestHelper, ns string, curlCommand string) {
+	retry.UntilSuccess(t, func(t test.TestHelper) {
+		oc.Exec(t,
+			pod.MatchingSelector("app=sleep", ns),
+			"sleep",
+			curlCommand,
+			assert.OutputContains(
+				"403",
+				"Got the expected 403 Forbidden response",
+				"Expected the JWT Authorization Policy to reject request (expected HTTP status 403), but got a different HTTP code"))
 	})
 }
+
+func assertRequestJWTDenied_401(t test.TestHelper, ns string, curlCommand string) {
+	retry.UntilSuccess(t, func(t test.TestHelper) {
+		oc.Exec(t,
+			pod.MatchingSelector("app=sleep", ns),
+			"sleep",
+			curlCommand,
+			assert.OutputContains(
+				"401",
+				"Got the expected 401 Unauthorized response",
+				"Expected the JWT AuthorizationPolicy to reject request (expected HTTP status 401), but got a different HTTP code"))
+	})
+}
+
+func assertRequestJWTAccepted(t test.TestHelper, ns string, curlCommand string) {
+	retry.UntilSuccess(t, func(t test.TestHelper) {
+		oc.Exec(t,
+			pod.MatchingSelector("app=sleep", ns),
+			"sleep",
+			curlCommand,
+			assert.OutputContains(
+				"200",
+				"Got the expected 200 OK response for the JWT request",
+				"Expected the JWT AuthorizationPolicy to accept request (expected HTTP status 200), but got a different HTTP code"))
+	})
+}
```

Comment:
Delete this and call `assertRequestAccepted()` in `deny_test.go`.

Since the assert functions in `deny_test.go` will now be used in two different files, it might make sense to move them to a separate file like `common.go` (in the same package).